app/models/item.rb

```diff
@@ -0,0 +1,14 @@
+class Item < ApplicationRecord
+  belongs_to :category
+  validates_presence_of :name
+  validates_uniqueness_of :name
+
+  before_validation :downcase_name
+
+  def downcase_name
+    if self.name?
```

Comment:
I would change the `before_validation` to `before_save`. Validating an item doesn't depend on it's name being in downcase, all that matters is that it has a name. What matters is that when it is saved it's name is downcased.